receiver/prometheusreceiver/internal/starttimemetricadjuster.go

```diff
@@ -15,25 +18,55 @@ var (
 	errNoStartTimeMetrics             = errors.New("start_time metric is missing")
 	errNoDataPointsStartTimeMetric    = errors.New("start time metric with no data points")
 	errUnsupportedTypeStartTimeMetric = errors.New("unsupported data type for start time metric")
+
+	// collectorStartTime is the approximate start time of the collector. Used
+	// as a fallback start time for metrics that don't have a start time set.
+	// Set when the component is initialized.
+	collectorStartTime *time.Time
 )
 
+func init() {
+	p, err := process.NewProcess(int32(os.Getpid()))
+	if err != nil {
+		panic("Unable to find current process" + err.Error())
+	}
+
+	ct, err := p.CreateTime()
+	if err != nil {
+		panic("Unable to find process start time" + err.Error())
+	}
+
+	startTime := time.Unix(ct/1000, 0) // Convert milliseconds to seconds
+	collectorStartTime = &startTime
+}
+
 type startTimeMetricAdjuster struct {
 	startTimeMetricRegex *regexp.Regexp
+	fallbackStartTime    *time.Time
 	logger               *zap.Logger
 }
 
 // NewStartTimeMetricAdjuster returns a new MetricsAdjuster that adjust metrics' start times based on a start time metric.
-func NewStartTimeMetricAdjuster(logger *zap.Logger, startTimeMetricRegex *regexp.Regexp) MetricsAdjuster {
+func NewStartTimeMetricAdjuster(logger *zap.Logger, startTimeMetricRegex *regexp.Regexp, useCollectorStartTimeFallback bool) MetricsAdjuster {
+	var fallbackStartTime *time.Time
+	if useCollectorStartTimeFallback {
+		fallbackStartTime = collectorStartTime
+	}
 	return &startTimeMetricAdjuster{
 		startTimeMetricRegex: startTimeMetricRegex,
+		fallbackStartTime:    fallbackStartTime,
 		logger:               logger,
 	}
 }
 
 func (stma *startTimeMetricAdjuster) AdjustMetrics(metrics pmetric.Metrics) error {
 	startTime, err := stma.getStartTime(metrics)
 	if err != nil {
-		return err
+		if stma.fallbackStartTime == nil {
```

Comment:
You can just evaluate the feature gate here, rather than plumbing booleans through the code base.